resources/tests/org.eclipse.core.tests.resources/src/org/eclipse/core/tests/resources/regression/Bug_297635.java

```diff
@@ -0,0 +1,147 @@
+/*******************************************************************************
+ *  Copyright (c) 2010, 2015 IBM Corporation and others.
+ *
+ *  This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License 2.0
+ *  which accompanies this distribution, and is available at
+ *  https://www.eclipse.org/legal/epl-2.0/
+ *
+ *  SPDX-License-Identifier: EPL-2.0
+ *
+ *  Contributors:
+ *     IBM Corporation - initial API and implementation
+ *     Alexander Kurtakov <akurtako@redhat.com> - Bug 459343
+ *******************************************************************************/
+package org.eclipse.core.tests.resources.regression;
+
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.net.MalformedURLException;
+import java.util.function.Consumer;
+import org.eclipse.core.internal.resources.SaveManager;
+import org.eclipse.core.resources.ISaveContext;
+import org.eclipse.core.resources.ISaveParticipant;
+import org.eclipse.core.resources.IWorkspace;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.core.tests.harness.BundleTestingHelper;
+import org.eclipse.core.tests.resources.ResourceTest;
+import org.eclipse.core.tests.resources.content.ContentTypeTest;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.BundleException;
+
+/**
+ * Tests regression of bug 297635
+ */
+public class Bug_297635 extends ResourceTest {
+	private Bundle saveParticipantBundle;
+
+	@Override
+	protected void setUp() throws Exception {
+		super.setUp();
+		saveParticipantBundle = SaveParticipant.installContainingBundleAndRegister();
+		saveFull();
+	}
+
+	@Override
+	protected void tearDown() throws Exception {
+		if (saveParticipantBundle != null) {
```

Comment:
Could this condition ever be `false` ? If `saveParticipantBundle == null` then there was an exception in the `setUp` method and the `tearDown` wouldn't run, or am I missing something?

Reply:
Sorry, I do not understand that comment. Why should `tearDown` not be called in case `setUp` throws an exception? This condition is _exactly_ for the case that `setUp` failed due to some exception.

Reply:
I was not aware of the unexpected JUnit 3 behavior of not executing `tearDown()` when `setUp()` throws an exception. I consider this an actual flaw as it breaks test isolation. I have documented the behavior in #742.

Concerning this PR, we should keep the check during `tearDown()`, as we hopefully fix the behavior for exceptions during `setUp()` by JUnit migration or a JUnit 3 workaround.

Reply:
> I was not aware of the unexpected **JUnit 3** behavior of not executing tearDown() when setUp() throws an exception. I consider this an actual flaw as it breaks test isolation.

I think JUnit3 is really not something to recommend these days, so worry about incomplete / inadequate features seems not help much and better instead move on to JUnit 5 (or 4 at least).

Reply:
> I think JUnit3 is really not something to recommend these days, so worry about incomplete / inadequate features seems not help much and better instead move on to JUnit 5 (or 4 at least).

Totally agree. Junit 3 should never be used to implement new tests (if possible). Here, it's simply legacy, as several tests (including the one adapted in this PR) are still based on JUnit 3 and it seems to be quite some effort to properly migrate them, We had a discussion about that some months ago (I think related to some addition of `assume` statements), but right now I do cannot find it anymore.

Reply:
To be clear: I liked the code better before _i.e._ even though `tearDown` is not being called with the current implementation (because it's a mix between JUnit3 and JUnit4), the check would make sense in JUnit5. I propose you leave the null-check in place so this test will be "future-ready" :-)

Reply:
Yes, that was an unintended change :-)
I adapted the behavior in 19dbe22.